### PR TITLE
Adding 3 new routes for manifest retrieval/upload, and some helper functions

### PR DESCRIPTION
**Pull request by piotrsenkow**
…nctions + formatting with python black

Link to JIRA ticket if there is one: 
https://ctds-planx.atlassian.net/browse/HP-1455

### New Features
Added 3 new endpoints:
- POST /metadata, allows to POST a json object containing discovery metadata for a given study, returns file name stored in s3 bucket. 
- GET /metadata, List all exported metadata objects associated with user
- GET /metadata/<filename.json>, Read the contents of an exported metadata object in the user's folder

### CODE REVIEW

File: manifestservice/manifests/__init__.py [hunk 1]
```diff
@@ -354,7 +496,10 @@ def _list_files_in_bucket(bucket_name, folder):
         "cohorts": [
             # For files in the cohorts/ folder
             { "filename": <filename>, "last_modified": <timestamp> }, ...
-        ]
+        ],
+        "metadata": [
+            { "external_oidc_idp": "qdr-keycloak", "file_retriever": "QDR", "study_id": "doi:10.5064/F6N2GOC9" }, ...
```
Comment (mfshao) [inline manifestservice/manifests/__init__.py:501]:
```suggestion
            # For files in the exported-metadata/ folder
            { "filename": <filename>, "last_modified": <timestamp> }, ...
```
I believe this is what it suppose to return?

File: manifestservice/manifests/__init__.py [hunk 2]
```diff
@@ -305,9 +439,7 @@ def is_valid_manifest(manifest_json, required_keys):
     return True
 
 
-def _generate_unique_manifest_filename(
-    folder_name, manifest_bucket_name, users_existing_manifest_files
-):
+def _generate_unique_manifest_filename(users_existing_manifest_files):
```
Comment (mfshao) [inline manifestservice/manifests/__init__.py:442]: I'm finding this confusing. I understand that in the master branch that the first two arguments that got passed into this function is not being used, so removing them makes sense. But then you should also change the signature at where this function got called at https://github.com/uc-cdis/manifestservice/pull/43/files#diff-fabdcc5ef3a4225800a012be5195c800e5d318dd779a5694b68dcaa230632fb1R360

File: manifestservice/manifests/__init__.py [hunk 3]
```diff
@@ -320,6 +452,15 @@ def _generate_unique_manifest_filename(
     return filename
 
 
+def _generate_unique_metadata_filename(users_existing_metadata_files):
```
Comment (mfshao) [inline manifestservice/manifests/__init__.py:455]: This function looks exactly the same as `_generate_unique_manifest_filename()` above, I think we can consolidate them by given it a more generalized name

**Review by mfshao — COMMENTED** [review]
There are many `TODO`s in the comments and I feel that most (if not all) of them has already been addressed? If they are, then let's clean them up

File: manifestservice/manifests/__init__.py (hunk 3 above)
Comment (piotrsenkow) [reply manifestservice/manifests/__init__.py]: Hi @mfshao I have addresses all your recent comments, thank you for the code review

File: manifestservice/manifests/__init__.py [hunk 4]
```diff
@@ -305,15 +434,17 @@ def is_valid_manifest(manifest_json, required_keys):
     return True
 
 
-def _generate_unique_manifest_filename(
-    folder_name, manifest_bucket_name, users_existing_manifest_files
+def _generate_unique_manifest_or_metadata_filename(
```
Comment (mfshao) [inline manifestservice/manifests/__init__.py:437]:
```suggestion
def _generate_unique_filename(
```

**Comment by mfshao** [comment]
code-wise lgtm, let's fill out the PR template with proper description about what this PR is doing, and bump up the version number in the porty config file

**Comment by mfshao** [comment]
oh and also, can you add some proper unit tests for this new feature? Just like what we have for those manifest and cohorts in the `/tests` directory?

**Comment by piotrsenkow** [comment]
> oh and also, can you add some proper unit tests for this new feature? Just like what we have for those manifest and cohorts in the `/tests` directory?

@mfshao I believe I finally got the unit tests to pass, could you please confirm? If necessary I can add more cases however this seemed like a good starting point.

File: tests/metadata_test.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import json as json_utils
+
+
+def test_POST_successful_metadata_add(client, mocks):
+    """
+        Test the Export PFB to Workspace pathway: a cohort is added to the bucket.
```
Comment (mfshao) [inline tests/metadata_test.py:6]:
```suggestion
        Test the export metadata to Workspace pathway: a metadata file is added to the bucket.
```

**Review by mfshao — COMMENTED** [review]
some small veribage change, the rest lgtm

**Review by mfshao — APPROVED** [review]
👍 great work!

**Pull request merged**